src/submap.h

```diff
@@ -418,6 +418,35 @@ class maptile_impl
             return *std::prev( sm->get_items( pos() ).cend() );
         }
 
+        /** Returns the uppermost favorite item if there is a favorite item. Otherwise, returns the item with the greatest volume.
+        *
+        * Assumes there is at least one item.
```

Comment:
You should check it in the code instead of silently returning nullptr. Throw some cata error (not sure how from the top of my head).

Reply:
I put that there because get_uppermost_item() made that assumption, so I figured I could too. I'll change it though.